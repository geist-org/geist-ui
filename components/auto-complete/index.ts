import AutoComplete from './auto-complete'
import AutoCompleteItem from './auto-complete-item'
import AutoCompleteSearching from './auto-complete-searching'
import AutoCompleteEmpty from './auto-complete-empty'

export type AutoCompleteComponentType = typeof AutoComplete & {
  Item: typeof AutoCompleteItem
  Option: typeof AutoCompleteItem
  Searching: typeof AutoCompleteSearching
  Empty: typeof AutoCompleteEmpty
}
;(AutoComplete as AutoCompleteComponentType).Item = AutoCompleteItem
;(AutoComplete as AutoCompleteComponentType).Option = AutoCompleteItem
;(AutoComplete as AutoCompleteComponentType).Searching = AutoCompleteSearching
;(AutoComplete as AutoCompleteComponentType).Empty = AutoCompleteEmpty

export type {
  AutoCompleteOption,
  AutoCompleteOptions,
  AutoCompleteProps,
  AutoCompleteTypes,
} from './auto-complete'
export default AutoComplete as AutoCompleteComponentType
