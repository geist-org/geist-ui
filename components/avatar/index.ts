import Avatar from './avatar'
import AvatarGroup from './avatar-group'

export type AvatarComponentType = typeof Avatar & {
  Group: typeof AvatarGroup
}
;(Avatar as AvatarComponentType).Group = AvatarGroup

export type { AvatarProps } from './avatar'
export type { AvatarGroupProps } from './avatar-group'

export default Avatar as AvatarComponentType
