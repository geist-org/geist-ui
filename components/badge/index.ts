import Badge from './badge'
import BadgeAnchor from './badge-anchor'

export type BadgeComponentType = typeof Badge & {
  Anchor: typeof BadgeAnchor
}
;(Badge as BadgeComponentType).Anchor = BadgeAnchor

export type { BadgeProps, BadgeTypes } from './badge'
export type { BadgeAnchorProps, BadgeAnchorPlacement } from './badge-anchor'
export default Badge as BadgeComponentType
