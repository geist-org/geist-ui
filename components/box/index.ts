import Box from './box'

export type { BoxProps } from './box'
export default Box
