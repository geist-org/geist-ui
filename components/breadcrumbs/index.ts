import Breadcrumbs from './breadcrumbs'
import BreadcrumbsItem from './breadcrumbs-item'
import BreadcrumbsSeparator from './breadcrumbs-separator'

export type BreadcrumbsComponentType = typeof Breadcrumbs & {
  Item: typeof BreadcrumbsItem
  Separator: typeof BreadcrumbsSeparator
}
;(Breadcrumbs as BreadcrumbsComponentType).Item = BreadcrumbsItem
;(Breadcrumbs as BreadcrumbsComponentType).Separator = BreadcrumbsSeparator

export type { BreadcrumbsProps } from './breadcrumbs'
export type { BreadcrumbsItemProps } from './breadcrumbs-item'
export type { BreadcrumbsSeparatorProps } from './breadcrumbs-separator'
export default Breadcrumbs as BreadcrumbsComponentType
