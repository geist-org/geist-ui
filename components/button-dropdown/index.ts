import ButtonDropdown from './button-dropdown'
import ButtonDropdownItem from './button-dropdown-item'

type ButtonDropdownType = typeof ButtonDropdown & {
  Item: typeof ButtonDropdownItem
}
;(ButtonDropdown as ButtonDropdownType).Item = ButtonDropdownItem

export type { ButtonDropdownProps, ButtonDropdownTypes } from './button-dropdown'
export type {
  ButtonDropdownItemProps,
  ButtonDropdownItemTypes,
} from './button-dropdown-item'
export default ButtonDropdown as ButtonDropdownType
