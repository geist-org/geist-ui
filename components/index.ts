// @ts-ignore
/// <reference types="styled-jsx" />

export { default as AutoComplete } from './auto-complete'
export type { AutoCompleteProps } from './auto-complete'

export { default as Avatar } from './avatar'
export type { AvatarProps, AvatarGroupProps } from './avatar'

export { default as Badge } from './badge'
export type { BadgeProps, BadgeAnchorProps } from './badge'

export { default as Breadcrumbs } from './breadcrumbs'
export type {
  BreadcrumbsProps,
  BreadcrumbsItemProps,
  BreadcrumbsSeparatorProps,
} from './breadcrumbs'

export { default as Button } from './button'
export type { ButtonProps } from './button'

export { default as ButtonDropdown } from './button-dropdown'
export type { ButtonDropdownProps, ButtonDropdownItemProps } from './button-dropdown'

export { default as ButtonGroup } from './button-group'
export type { ButtonGroupProps } from './button-group'

export { default as Capacity } from './capacity'
export type { CapacityProps } from './capacity'

export { default as Card } from './card'
export type { CardProps, CardContentProps, CardFooterProps } from './card'

export { default as Checkbox } from './checkbox'
export type { CheckboxProps, CheckboxGroupProps } from './checkbox'

export { default as Code } from './code'
export type { CodeProps } from './code'

export { default as Collapse } from './collapse'
export type { CollapseProps, CollapseGroupProps } from './collapse'

export { default as Description } from './description'
export type { DescriptionProps } from './description'

export { default as Display } from './display'
export type { DisplayProps } from './display'

export { default as Divider } from './divider'
export type { DividerProps } from './divider'

export { default as Dot } from './dot'
export type { DotProps } from './dot'

export { default as Drawer } from './drawer'
export type { DrawerProps } from './drawer'

export { default as Fieldset } from './fieldset'
export type {
  FieldsetProps,
  FieldsetTitleProps,
  FieldsetSubtitleProps,
  FieldsetGroupProps,
  FieldsetFooterProps,
  FieldsetContentProps,
} from './fieldset'

export { default as GeistProvider } from './geist-provider'
export type { GeistProviderProps } from './geist-provider'

export { default as Grid } from './grid'
export type { GridProps, GridContainerProps } from './grid'

export { default as Image } from './image'
export type { ImageProps, ImageBrowserProps } from './image'

export { default as Input } from './input'
export type { InputProps, InputPasswordProps } from './input'

export { default as Keyboard } from './keyboard'
export type { KeyboardProps } from './keyboard'

export { default as Link } from './link'
export type { LinkProps } from './link'

export { default as Loading } from './loading'
export type { LoadingProps } from './loading'

export { default as Modal } from './modal'
export type {
  ModalProps,
  ModalTitleProps,
  ModalSubtitleProps,
  ModalContentProps,
  ModalActionProps,
} from './modal'

export { default as Note } from './note'
export type { NoteProps } from './note'

export { default as Page } from './page'
export type {
  PageProps,
  PageHeaderProps,
  PageContentProps,
  PageFooterProps,
} from './page'

export { default as Pagination } from './pagination'
export type {
  PaginationProps,
  PaginationNextProps,
  PaginationPreviousProps,
} from './pagination'

export { default as Popover } from './popover'
export type { PopoverProps, PopoverItemProps } from './popover'

export { default as Progress } from './progress'
export type { ProgressProps } from './progress'

export { default as Radio } from './radio'
export type { RadioProps, RadioGroupProps, RadioDescriptionProps } from './radio'

export { default as Rating } from './rating'
export type { RatingProps } from './rating'

export { default as Select } from './select'
export type { SelectProps, SelectOptionProps } from './select'

export { default as Slider } from './slider'
export type { SliderProps } from './slider'

export { default as Snippet } from './snippet'
export type { SnippetProps } from './snippet'

export { default as Spacer } from './spacer'
export type { SpacerProps } from './spacer'

export { default as Spinner } from './spinner'
export type { SpinnerProps } from './spinner'

export { default as Table } from './table'
export type { TableProps, TableColumnProps } from './table'

export { default as Tabs } from './tabs'
export type { TabsProps } from './tabs'

export { default as Tag } from './tag'
export type { TagProps } from './tag'

export { default as Text } from './text'
export type { TextProps } from './text'

export { default as Textarea } from './textarea'
export type { TextareaProps } from './textarea'

export { default as Themes } from './themes'
export type { GeistUIThemes, GeistUserTheme } from './themes'

export { default as Toggle } from './toggle'
export type { ToggleProps } from './toggle'

export { default as Tooltip } from './tooltip'
export type { TooltipProps } from './tooltip'

export { default as Tree } from './tree'
export type { TreeProps } from './tree'

export { default as useAllThemes } from './use-all-themes'
export type { AllThemesConfig } from './use-all-themes'

export { default as useToasts } from './use-toasts'
export type { Toast, ToastInput, ToastAction, ToastLayout } from './use-toasts'

export { default as User } from './user'
export type { UserProps } from './user'

export { default as useBodyScroll } from './use-body-scroll'
export type { BodyScrollOptions } from './use-body-scroll'

export { default as useClipboard } from './use-clipboard'
export type { UseClipboardOptions } from './use-clipboard'

export { default as useMediaQuery } from './use-media-query'
export type { ResponsiveOptions, ResponsiveBreakpoint } from './use-media-query'

export { default as useKeyboard, KeyMod, KeyCode } from './use-keyboard'
export type { KeyboardOptions, UseKeyboardHandler } from './use-keyboard'

export { default as useInput } from './use-input'
export { default as useModal } from './use-modal'
export { default as useTabs } from './use-tabs'
export { default as useClickAway } from './use-click-away'
export { default as useCurrentState } from './use-current-state'
export { default as CssBaseline } from './css-baseline'
export { default as useTheme } from './use-theme'
export { default as useClasses } from './use-classes'
export { default as useScale } from './use-scale'
export { withScale, ScalePropKeys, ScaleContext } from './use-scale'
export type {
  ScaleProps,
  ScaleConfig,
  GetScalePropsFunction,
  GetAllScalePropsFunction,
} from './use-scale'

export { useRect } from './utils/layouts'
export type { ReactiveDomReact } from './utils/layouts'
export { default as Highlight } from './shared/highlight'
export type { HighlightProps } from './shared/highlight'
