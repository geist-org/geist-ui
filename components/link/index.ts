import Link from './link'

export type { LinkProps } from './link'
export default Link
