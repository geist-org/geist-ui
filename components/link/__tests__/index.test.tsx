import React from 'react'
import { mount } from 'enzyme'
import { Link } from 'components'

describe('Link', () => {
  it('should render correctly', () => {
    const wrapper = mount(
      <div>
        <Link href="https://geist-ui.dev">link</Link>
        <Link href="https://geist-ui.dev" color>
          link
        </Link>
        <Link href="https://geist-ui.dev" icon>
          link
        </Link>
        <Link href="https://geist-ui.dev" underline>
          link
        </Link>
        <Link href="https://geist-ui.dev" block>
          link
        </Link>
      </div>,
    )
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('should be no errors when href missing', () => {
    const errorSpy = jest.spyOn(console, 'error')
    const wrapper = mount(<Link />)
    expect(errorSpy).not.toHaveBeenCalled()
    expect(() => wrapper.unmount()).not.toThrow()
    errorSpy.mockRestore()
  })

  it('should forward ref', () => {
    let ref = React.createRef<HTMLAnchorElement>()
    const errorSpy = jest.spyOn(console, 'error')
    const wrapper = mount(<Link ref={ref} />)

    expect(errorSpy).not.toHaveBeenCalled()
    expect(ref.current).not.toBeNull()
    expect(() => wrapper.unmount()).not.toThrow()
    errorSpy.mockRestore()
  })

  it('an warning should be thrown when using the pure prop', () => {
    const errorSpy = jest.spyOn(console, 'error').mockImplementation(() => {})
    mount(<Link pure />)
    expect(errorSpy).toHaveBeenCalled()
    errorSpy.mockRestore()
  })
})
