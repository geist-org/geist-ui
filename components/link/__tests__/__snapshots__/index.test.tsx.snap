// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render correctly 1`] = `
"<div><a class=\\"link\\" href=\\"https://geist-ui.dev\\">link<style>
          .link {
            display: inline-flex;
            align-items: baseline;
            line-height: inherit;
            color: inherit;
            text-decoration: none;
            border-radius: 0;
            transition: color 200ms ease 0ms;
            font-size: inherit;
            width: fit-content;
            height: auto;
            margin: 0 0 0 0;
            padding: 0 0 0 0;
          }
          .block {
            padding: calc(0.125 * 16px) calc(0.25 * 16px) calc(0.125 * 16px)
              calc(0.25 * 16px);
            margin: 0 calc(-0.125 * 16px) 0
              calc(-0.125 * 16px);
          }

          .link:hover,
          .link:active,
          .link:focus {
            text-decoration: none;
          }

          .link:hover {
            background-color: unset;
            color: inherit;
          }
        </style></a><a class=\\"link\\" href=\\"https://geist-ui.dev\\">link<style>
          .link {
            display: inline-flex;
            align-items: baseline;
            line-height: inherit;
            color: #0070f3;
            text-decoration: none;
            border-radius: 0;
            transition: color 200ms ease 0ms;
            font-size: inherit;
            width: fit-content;
            height: auto;
            margin: 0 0 0 0;
            padding: 0 0 0 0;
          }
          .block {
            padding: calc(0.125 * 16px) calc(0.25 * 16px) calc(0.125 * 16px)
              calc(0.25 * 16px);
            margin: 0 calc(-0.125 * 16px) 0
              calc(-0.125 * 16px);
          }

          .link:hover,
          .link:active,
          .link:focus {
            text-decoration: none;
          }

          .link:hover {
            background-color: unset;
            color: #3291ff;
          }
        </style></a><a class=\\"link\\" href=\\"https://geist-ui.dev\\">link<svg viewBox=\\"0 0 24 24\\" width=\\"0.9375em\\" height=\\"0.9375em\\" stroke=\\"currentColor\\" stroke-width=\\"1.5\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" fill=\\"none\\" shape-rendering=\\"geometricPrecision\\" class=\\"icon\\"><path d=\\"M18 13v6a2 2 0 01-2 2H5a2 2 0 01-2-2V8a2 2 0 012-2h6\\"></path><path d=\\"M15 3h6v6\\"></path><path d=\\"M10 14L21 3\\"></path><style>
        .icon {
          margin: 0 0 -1px 0.1875em;
          display: inline-flex;
          align-self: center;
          color: currentColor;
        }
      </style></svg><style>
          .link {
            display: inline-flex;
            align-items: baseline;
            line-height: inherit;
            color: inherit;
            text-decoration: none;
            border-radius: 0;
            transition: color 200ms ease 0ms;
            font-size: inherit;
            width: fit-content;
            height: auto;
            margin: 0 0 0 0;
            padding: 0 0 0 0;
          }
          .block {
            padding: calc(0.125 * 16px) calc(0.25 * 16px) calc(0.125 * 16px)
              calc(0.25 * 16px);
            margin: 0 calc(-0.125 * 16px) 0
              calc(-0.125 * 16px);
          }

          .link:hover,
          .link:active,
          .link:focus {
            text-decoration: none;
          }

          .link:hover {
            background-color: unset;
            color: inherit;
          }
        </style></a><a class=\\"link\\" href=\\"https://geist-ui.dev\\">link<style>
          .link {
            display: inline-flex;
            align-items: baseline;
            line-height: inherit;
            color: inherit;
            text-decoration: none;
            border-radius: 0;
            transition: color 200ms ease 0ms;
            font-size: inherit;
            width: fit-content;
            height: auto;
            margin: 0 0 0 0;
            padding: 0 0 0 0;
          }
          .block {
            padding: calc(0.125 * 16px) calc(0.25 * 16px) calc(0.125 * 16px)
              calc(0.25 * 16px);
            margin: 0 calc(-0.125 * 16px) 0
              calc(-0.125 * 16px);
          }

          .link:hover,
          .link:active,
          .link:focus {
            text-decoration: underline;
          }

          .link:hover {
            background-color: unset;
            color: inherit;
          }
        </style></a><a class=\\"link block\\" href=\\"https://geist-ui.dev\\">link<style>
          .link {
            display: inline-flex;
            align-items: baseline;
            line-height: inherit;
            color: #0070f3;
            text-decoration: none;
            border-radius: 6px;
            transition: color 200ms ease 0ms;
            font-size: inherit;
            width: fit-content;
            height: auto;
            margin: 0 0 0 0;
            padding: 0 0 0 0;
          }
          .block {
            padding: calc(0.125 * 16px) calc(0.25 * 16px) calc(0.125 * 16px)
              calc(0.25 * 16px);
            margin: 0 calc(-0.125 * 16px) 0
              calc(-0.125 * 16px);
          }

          .link:hover,
          .link:active,
          .link:focus {
            text-decoration: none;
          }

          .link:hover {
            background-color: rgba(0, 112, 243, 0.1);
            color: #3291ff;
          }
        </style></a></div>"
`;
